File: Web/src/app/core/models/diaper-log.ts
import { TemplateRef } from "@angular/core"

export interface DiaperTrackerData {
  diaperType: string
  peeColor: string
  stoolColor: string[]
  stoolTexture: string[]
  rash: string
  rashLocation: string[]
  startTime: string
  notes: string
}

export interface DiaperStep {
  name: string
  templateRef?: TemplateRef<any>
}

File: Api/src/MilkAndMoon.Api/Models/DiaperLog.cs
namespace MilkAndMoon.Api.Models;

public class DiaperLog : BabyLog
{
    public string DiaperType { get; set; } = null!;
    public string? PeeColor { get; set; }
    public string[] StoolColor { get; set; } = null!;
    public string[] StoolTexture { get; set; } = null!;
    public string? Rash { get; set; }
    public string[] RashLocation { get; set; } = null!;
}
